hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ResourceManager.java

```diff
@@ -824,25 +824,29 @@ public void handle(RMFatalEvent event) {
    * Transition to standby in a new thread.
    */
   public void handleTransitionToStandByInNewThread() {
-    new Thread() {
-      @Override
-      public void run() {
-        if (rmContext.isHAEnabled()) {
-          try {
-            // Transition to standby and reinit active services
-            LOG.info("Transitioning RM to Standby mode");
-            transitionToStandby(true);
-            EmbeddedElector elector = rmContext.getLeaderElectorService();
-            if (elector != null) {
-              elector.rejoinElection();
-            }
-          } catch (Exception e) {
-            LOG.fatal("Failed to transition RM to Standby mode.", e);
-            ExitUtil.terminate(1, e);
+    Thread standByTransitionThread = new Thread(new StandByTransitionThread());
+    standByTransitionThread.setName("StandByTransitionThread Handler");
+    standByTransitionThread.start();
+  }
+
+  private class StandByTransitionThread implements Runnable {
```

Comment:
Naming the Runnable a Thread sounds confusing. Can we change it to TransitionToStandbyRunnable or some such?